### ISSUE TITLE
Create a GitHub Action to Publish 

### ISSUE THREAD

**Issue by pjhill** [department-of-veterans-affairs/va.gov-team#41295]
# Description
Alter the existing GitHub Action workflow in the [product-directory](https://github.com/department-of-veterans-affairs/product-directory) that uploads our product list to BigQuery, to add steps that will publish the contents of the updated product list to the primary Confluence location. Additionally, change the trigger from a once-a-day run to when a pull request to update the list has been merged.

# Tasks
- [x] Update existing GHA workflow to include new steps to publish to Confluence
- [x] Update the workflow trigger from once-a-day to run when pull requests are merged.
- [x] Test the workflow with pull requests
- [x] Verify publication to confluence is successful
- [x] Establish notifications for failed publications in the #qas-notifications channel

# Acceptance Criteria
- [ ] An event-driven GitHub Actions workflow runs in the product directory repository when a pull request to update the product list is merged
- [ ] When changes to the product list are merged, the updated list is published to Confluence

**Comment by rsmithadhoc** [comment]
- Opened a [pull request](https://github.com/department-of-veterans-affairs/product-directory/pull/175) with additional changes.
- **If someone wants to pick this up while I am out of the office:**
  - This is just waiting on a Confluence email and token. I updated the process to use those environment variables.
  - I believe all that is needed is updating `.github/workflows/continuous-integration.yml` to pull those values from the parameter store once they are there.
  - To test this, you can update `.github/workflows/continuous-integration.yml` and comment out the `if:` under `publish-product-directory`, then open a PR to run that step and confirm it publishes. Be sure to uncomment the `if:` when done so that this step only runs on a merge to `main`.

**Comment by pjhill** [comment]
@rsmithadhoc -- What do you think about launching with an individual user's token in place? I suppose that still leaves the challenge of getting the token into AWS parameter store, but I believe we have the permission to create and update AWS param store with what we need, right @rbeckwith-oddball ?

**Comment by rsmithadhoc** [comment]
@pjhill It's working with my token, I put links to the successful test run with testing instructions on https://github.com/department-of-veterans-affairs/product-directory/pull/175#issue-1256711763 I think that PR is ready for a review then. 🎉

**Comment by pjhill** [comment]
The ability to publish to multiple Confluence pages with different subsets of data displayed is not part of the scope of this ticket. I'll make a new ticket to accomplish that.

**Comment by pjhill** [comment]
PR is approved and ready to merge, so I'm closing this out.

**Comment by rsmithadhoc** [comment]
> This is looking pretty great! I think once we get this PR merged we can iterate on this a bit to add the capability of generating the publish content based on a selection of KV pairs from the JSON. We may be headed toward a solution where we publish different subsets of the JSON data to different Confluence pages.
> 
> Two of the initial use cases we are solving for are --
> 
>     1. Publish all the content in the JSON
> 
>     2. Publish just the content from the JSON that the Governance Team is already tracking in their Confluence page [here](https://vfs.atlassian.net/wiki/spaces/VAProducts/overview).

Thanks for taking a look! That sounds good, it should be straightforward to include or exclude certain fields from the export. I already have it set up to publish multiple pages if those are configured.
